packages/test/src/FixtureManager.ts

```diff
@@ -29,7 +29,10 @@ export default class FixtureManager implements Manager {
             const { key } = action.meta;
 
             if (key in this.mockResults) {
-              dispatch(this.mockResults[key] as any);
+              // All updates must be async or React will complain about re-rendering in same pass
+              setTimeout(() => {
+                dispatch(this.mockResults[key] as any);
+              }, 0);
               return Promise.resolve();
```

Comment:
```suggestion
              return new Promise((resolve) => {
                resolve(dispatch(this.mockResults[key] as any));
              });
```
that way if it throws or returns a promise, it'll be caught/handled

Reply:
Yeah, got handling rejection in next commit. I'll just push that as part of this PR as well